com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/CopybookScanAnalysis.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *
+ *   Broadcom, Inc. - initial API and implementation
+ */
+package com.ca.lsp.cobol.service;
+
+import static com.ca.lsp.cobol.service.utils.FileSystemService.isFileUnderExtendedSourceFolder;
+
+/** This enum class used to track text synchronization type for the processed document. */
+public enum CopybookScanAnalysis {
+  ENABLED,
+  DISABLED;
```

Comment:
I'm not sure that you need this as a separate enum. You may just reuse the TextDocumentSyncType, but change its semantic: rename to something more descriptive, e.g. CopybookProcessingMode and redefine its elements to something kinda RESOLVE, IGNORE and SKIP. 
The RESOLVE will act as DID_OPEN now, IGNORE as DID_CHANGE (as I can see, we don't really check this value, only equity to DID_OPEN, but let it be), and SKIP will disable the copybook processing for the extended sources. 
So, you will call the `isFileUnderExtendedSourceFolder()` on `didOpen` or `didChange` and send the according value. The value is that the purpose of the parameter goes from its name, i.e. it is more descriptive. Also, you will have only one enum instead of two.
This is only an idea, the decision is up to you

Reply:
Initially I defined one enum and trowed away did_open, change and so on. But later on I verified that we still need to distinguish if a document is in did_open or did_change. Because there are two conditions that will disable the copybook analysis: 
- the file is an extended document
- the file is in did_change
So for that reason I decided to separate those two concerns into two enums.
Btw I like the name you suggested ```CopybookProcessingMode``` and the values, so if u are agree I will rename the enum ```CopybookScanAnalysis``` that didn't like a lot as name.

Reply:
Ok, why not

Reply:
💪